### PR TITLE
feat: new needed structs

### PR DESCRIPTION
**Pull request by boianradu**

### CODE REVIEW

File: v1/submissions.go [hunk 1]
```diff
@@ -250,3 +474,54 @@ func (e *Endpoint) GetSubmissions() ([]Submission, error) {
 
 	return findings, nil
 }
+
+/*
+	GetSpecificSubmission gets report/submission by code/id
+*/
+func (e *Endpoint) GetSpecificSubmission(code string) (SubmissionComplete, error) {
```
Comment (emilva) [inline v1/submissions.go]: `GetSubmission`

File: v1/submissions.go [hunk 2]
```diff
@@ -144,6 +145,229 @@ type Submission struct {
 	CloseReason string
 }
 
+/*
+	INTIGRITI Report structure
+*/
+type SubmissionComplete struct {
```
Comment (emilva) [inline v1/submissions.go]: `SubmissionDetails`

File: v1/submissions.go [hunk 3]
```diff
@@ -250,3 +474,54 @@ func (e *Endpoint) GetSubmissions() ([]Submission, error) {
...
+
+/*
+	GetSpecificSubmission gets report/submission by code/id
+*/
+func (e *Endpoint) GetSpecificSubmission(code string) (SubmissionComplete, error) {
+	var submi SubmissionComplete
+	var respBytes []byte
+	var err error
+	var req *http.Request
+	url := "https://api.intigriti.com/external/v1.2/submissions/" + code
```
Comment (emilva) [inline v1/submissions.go]: Update base url definition in `Endpoint.go`

File: v1/submissions.go [hunk 4]
```diff
@@ -250,3 +474,54 @@ func (e *Endpoint) GetSubmissions() ([]Submission, error) {
...
+
+	req, err = http.NewRequest(http.MethodGet, url, nil)
```
Comment (emilva) [inline v1/submissions.go:493]: Do the `if err := authenticate(e);` check before the request, as in `GetSubmissions()`

File: v1/submissions.go [hunk 5]
```diff
@@ -250,3 +474,54 @@ func (e *Endpoint) GetSubmissions() ([]Submission, error) {
...
+	if err != nil {
+		return SubmissionComplete{}, errors.Wrap(err, "could not create http request to intigriti")
```
Comment (emilva) [inline v1/submissions.go]: return `nil` instead of an empty struct for all these errors

File: v1/submissions.go [hunk 6]
```diff
@@ -250,3 +474,54 @@ func (e *Endpoint) GetSubmissions() ([]Submission, error) {
...
+	if resp.StatusCode > 399 {
+		return SubmissionComplete{}, errors.Errorf("fetch from intigriti returned status code: %d", resp.StatusCode)
+	}
+
+	respBytes, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return SubmissionComplete{}, errors.Wrap(err, "could not read response")
+	}
+	if err := json.Unmarshal(respBytes, &submi); err != nil {
+		return submi, errors.Wrap(err, "could not decode intigriti response")
```
Comment (emilva) [inline v1/submissions.go]: `submi` is never returned when it is successful, I guess that should be on the line below 👇

File: v1/submissions.go [hunk 7]
```diff
@@ -250,3 +474,54 @@ func (e *Endpoint) GetSubmissions() ([]Submission, error) {
...
+	}
+	return SubmissionComplete{}, errors.New("ends in error")
```
Comment (emilva) [inline v1/submissions.go]: should not end in error, but return the submission

File: v1/submissions.go (hunk 1 above)
Comment (boianradu) [reply v1/submissions.go]: ✅

File: v1/submissions.go (hunk 2 above)
Comment (boianradu) [reply v1/submissions.go]: ✅

File: v1/submissions.go (hunk 4 above)
Comment (boianradu) [reply v1/submissions.go:493]: ✅

File: v1/submissions.go (hunk 3 above)
Comment (boianradu) [reply v1/submissions.go]: ✅

File: v1/submissions.go (hunk 5 above)
Comment (boianradu) [reply v1/submissions.go]: ✅

File: v1/submissions.go (hunk 6 above)
Comment (boianradu) [reply v1/submissions.go]: ✅

File: v1/submissions.go (hunk 7 above)
Comment (boianradu) [reply v1/submissions.go]: ✅

**Pull request closed without merging**